lib/stdlib/test/lists_SUITE.erl

```diff
@@ -163,7 +163,7 @@ append_trapping_1(N) ->
     List = lists:duplicate(N + (1 bsl N), gurka),
     ImproperList = List ++ crash,
 
-    {'EXIT',_} = (catch (ImproperList ++ [])),
+    {'EXIT',_} = (catch (ImproperList ++ [1, 2, 3])),
```

Comment:
There's a behavior change here: with this patch we no longer check that the left hand side is a proper list. If the old behavior is desirable, should we scan through the left hand side in the new `if` branch above to check for an improper list?

Reply:
As far as I know, this is pretty much intentional. `[1, 2, 3] ++ Foo` is the same as `[1, 2, 3 | Foo]`. I think the correct fix is to, if an empty list is given, we check the left side to be a list or not, but never copy it (but it still raises if the list is improper). Better to wait for someone from the OTP team to confirm it though.

Reply:
Yes, it's intentional and we need to keep this behavior.